test-suite/mk-tls-test-certs.sh

```diff
@@ -0,0 +1,199 @@
+		organizationName= Example.Com
+		commonName= Test CA. Do not Trust
+		[ user_extensions ]
+		basicConstraints= CA:false
+	" >example.com-leaf
+
+	openssl genrsa --out leaf-rsa.pkey.tmp 4096
+
+	openssl req --new --sha256 --set_serial 3 \
+		--config example.com-leaf \
+		--key leaf-rsa.pkey.tmp \
+		--out leaf-rsa.csr
+
+	openssl rsa --in leaf-rsa.pkey.tmp --out leaf-rsa.pkey
+
+	openssl x509 --req --days 365 \
+		--in leaf-rsa.csr \
+		--CA ca-root-rsa.crt --CAkey ca-root-rsa.pkey --set_serial 3 \
+		--out leaf-rsa.crt
+
+	rm -f leaf-rsa.csr leaf-rsa.pkey.tmp example.com-leaf
+
+	#
+	# PEM files with CA chain
+	#
+	cat ca-root-rsa.pkey ca-root-rsa.crt > ca-root-rsa.pem
+	cat ca-mid-rsa.pkey ca-mid-rsa.crt > ca-mid-rsa.pem
+	cat ca-mid-rsa.pkey ca-mid-rsa.crt ca-root-rsa.crt > ca-chain-rsa.pem
+	cat leaf-rsa.pkey leaf-rsa.crt ca-root-rsa.crt > leaf-chain-rsa.pem
+	cat leaf-rsa.pkey leaf-rsa.crt > leaf-key-rsa.pem
+	cat leaf-rsa.crt ca-root-rsa.crt > leaf-chain-nokey-rsa.pem
+
+elif test "x$CERTTOOL" != "x" -a -x $CERTTOOL; then
+
+	#
+	# self-signed root CA
+	#
+	echo "
+		organization = \"Example.Org\"
+		cn = \"Test CA. Do not Trust\"
+		dc = \"example.org\"
+		expiration_days = 365
+		ca
+		cert_signing_key
+		crl_signing_key
+		" > example.org-ca
+
+	$CERTTOOL --generate-privkey --rsa --outfile ca-root-rsa.pkey
+
+	$CERTTOOL --generate-self-signed --template example.org-ca \
+		--rsa --load-privkey ca-root-rsa.pkey --outfile ca-root-rsa.crt
+
+	rm example.org-ca
+
+	#
+	# intermediary CA, signed by root
+	#
+	echo "
+	organization = \"Example.Net\"
+	cn = \"Test CA. Do not Trust\"
+	dc = \"example.net\"
+	expiration_days = 365
+	ca
+	cert_signing_key
+	crl_signing_key
+	" > example.net-ca
+
+	$CERTTOOL --generate-privkey --rsa --outfile ca-mid-rsa.pkey
+
+	$CERTTOOL --generate-certificate --load-privkey ca-mid-rsa.pkey \
+		--load-ca-certificate ca-root-rsa.crt --load-ca-privkey ca-root-rsa.pkey \
+		--template example.net-ca --outfile ca-mid-rsa.crt
+
+	rm example.net-ca
+
+	#
+	# Standard leaf / non-CA certificate
+	#
+	echo "
+	organization = \"Example.Com\"
+	cn = \"Test CA. Do not Trust\"
+	dc = \"example.com\"
+	expiration_days = 365
+	tls_www_server
+	" >example.com-leaf
+
+	$CERTTOOL --generate-privkey --rsa --outfile leaf-rsa.pkey
+
+	$CERTTOOL --generate-certificate --load-privkey leaf-rsa.pkey \
+		--load-ca-certificate ca-root-rsa.crt --load-ca-privkey ca-root-rsa.pkey \
+		--template example.com-leaf --outfile leaf-rsa.crt
+
+	rm -f example.com-leaf
+
+	#
+	# PEM files with CA chain
+	#
+	cat ca-root-rsa.pkey ca-root-rsa.crt > ca-root-rsa.pem
+	cat ca-mid-rsa.pkey ca-mid-rsa.crt > ca-mid-rsa.pem
+	cat ca-mid-rsa.pkey ca-mid-rsa.crt ca-root-rsa.crt > ca-chain-rsa.pem
+	cat leaf-rsa.pkey leaf-rsa.crt ca-root-rsa.crt > leaf-chain-rsa.pem
+	cat leaf-rsa.pkey leaf-rsa.crt > leaf-key-rsa.pem
+	cat leaf-rsa.crt ca-root-rsa.crt > leaf-chain-nokey-rsa.pem
+
+else
+	echo "WARNING: cannot find a tool to generate certificates"
```

Comment:
@yadij, you have marked this (and several other) change requests as resolved _and_ requested my review, but I do not see the corresponding changes (as of commit 59be2b4). Please double check the current state of the PR branch and re-request review when everything is ready.

FWIW, to reduce confusion, I recommend _not_ resolving other's change requests (at least until the corresponding changes are pushed/published) and adding (to the change request thread) a brief comment with the SHA of the commit addressing the change.

Reply:
github itself is doing the closure whenever suggested changes are merged. I accepted a lot of your suggestions as-is and now awaiting an update to your review status with those in, and other unresolved items responded to.

Reply:
> github itself is doing the closure whenever suggested changes are merged

I know. My recommendation still applies -- those GitHub actions can be undone when adding manual SHA comments (but I do not insist on that -- it is just a suggestion that would minimize confusion in cases like this change request).

> awaiting an update to your review status with those in

The problem is that, as I said, I do not _see_ those changes in the current PR code (as of commit 59be2b4). I have not researched when they were done, what happened to them since then, but that should not really matter. What matters is that they are not in the current PR (AFAICT).

Reply:
This is one of the change requests that was resolved unaddressed.

Reply:
Re-done in 1c45188